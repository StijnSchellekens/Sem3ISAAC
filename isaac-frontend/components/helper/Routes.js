import React from 'react';
import Link from 'next/link';

const Routes = () => {
  return (
    <div>
      <Link href='/dashboard'>
        <a>Dashboard</a>
      </Link>
      <Link href='/heatmap'>
        <a>Heatmap</a>
      </Link>
      <Link href='/notification'>
        <a>Notification</a>
      </Link>
      <Link href='/advanced'>
        <a>Advanced</a>
      </Link>
      <Link href='/settings'>
        <a>Settings</a>
      </Link>
    </div>
  );
};

export default Routes;
