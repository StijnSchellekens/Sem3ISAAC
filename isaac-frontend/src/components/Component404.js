import React from 'react';

const Component404 = () =>{

	return (
		<div>
            
		</div>
	);
};

export default Component404;