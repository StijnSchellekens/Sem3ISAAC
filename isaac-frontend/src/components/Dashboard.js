import React, {useState, useEffect} from 'react';
import AppBar from '@mui/material/AppBar';
import Toolbar from '@mui/material/Toolbar';
import Typography from '@mui/material/Typography';
import Gauge from '../components/Gauge';
import Grid from '@mui/material/Grid';
import DashboardGraphs from '../components/DashboardGraphs';
import Box from '@material-ui/core/Box';

const drawerWidth = 240;

const Dashboard = () => {
  const [data, setData] = useState(null);
  useEffect( async () => {
    const res = await fetch('http://localhost:5000/entries');
    const rawData = await res.json();

    setData(await rawData.map((obj) => {
      obj.dateTime = new Date(obj.dateTime);
      return obj;
    }));
  }, []);

  if (!data) {
    return (
      <div>...Loading</div>
    );
  }

  const generalStyle = {
    marginLeft: `${drawerWidth}px`,
    marginRight: `auto`,
  };

  const dashBoardStyle = {
    width: `calc(100% - ${drawerWidth}px)`,
  };

  const content = {
    marginLeft: `10%`,
    marginRight: `10%`,
  };

  // const halfPieChart = {
  //   marginLeft: 'auto',
  //   marginRigh: 'auto',
  // };

  return (
    <div style={generalStyle}>
      <AppBar
        position="fixed"
        color="primary"
        style={dashBoardStyle}
      >
        <Toolbar>
          <Typography variant="h6" noWrap component="div">
            Dashboard
          </Typography>
        </Toolbar>
      </AppBar>
      <Box style={content}>
        <Grid container
          spacing={2}
        >
          <Grid item xs={12} lg={6}>
            <Gauge id="graph-chart-temperature" name="Temperature" data={data}/>
          </Grid>
          <Grid item xs={12} lg={6}>
            <Gauge id="graph-chart-humidity" name="Humidity" data={data}/>
          </Grid>
          <Grid item xs={12}>
            <DashboardGraphs data={data}/>
          </Grid>
        </Grid>
      </Box>
    </div>
  );
};

export default Dashboard;
