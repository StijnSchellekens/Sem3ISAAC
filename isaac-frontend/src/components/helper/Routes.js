import React, {useState, useEffect} from 'react';


import Dashboard from '../Dashboard';
import Heatmap from '../Heatmap';
import Notifications from '../Notifications';
import Advanced from '../Advanced';
import Settings from '../Settings';

import {serverFetch} from '../../utils/server-fetch';

import {
	Switch,
	Route,
} from 'react-router-dom';

const Routes = () => {
	const [data, setData] = useState(null);
	useEffect( async () => {
		// const rawData = await (await fetch('http://localhost:5000/entries')).json();
		const rawData = await serverFetch();
		setData(await rawData.map((obj) => {
			obj.dateTime = new Date(obj.dateTime);
			return obj;
		}));
	}, []);

	return (
		<Switch>
			<Route path="/heatmap">
				<Heatmap data={data}/>
			</Route>
			<Route path='/notifications'>
				<Notifications />
			</Route>
			<Route path='/advanced'>
				<Advanced />
			</Route>
			<Route path='/settings'>
				<Settings />
			</Route>
			<Route path="/">
				<Dashboard data={data}/>
			</Route>
		</Switch>
	);
};

export default Routes;
