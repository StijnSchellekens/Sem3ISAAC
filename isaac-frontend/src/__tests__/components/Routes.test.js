/* eslint-disable no-undef */
/* eslint-disable no-mixed-spaces-and-tabs */
import React from 'react';
import { MemoryRouter } from 'react-router-dom';
import { shallow, render, mount, configure } from 'enzyme';
import Routes from '../../components/helper/Routes';

import Dashboard from '../../components/Dashboard';
import Component404 from '../../components/Component404';

const flushPromises = require('flush-promises');
import { waitForState } from 'enzyme-async-helpers';
import { createMemoryHistory } from 'history';
import toJSON from 'enzyme-to-json';
import ReactTestUtils from 'react-dom/test-utils';

// afterEach(() => {
// 	jest.resetModules();
// });

const mockFetch = Promise.resolve({
	json: () => Promise.resolve({
		entries:
        [
        	{
        		'id': 1,
        		'temp': 25.0,
        		'humidity': 45,
        		'dateTime': 'October 11, 2021 08:00:00',
        	},
        ],
	}),
});
global.fetch = jest.fn(() => mockFetch);


test('invalid paths should redirect to 404', async () => {
	fetch.mockImplementationOnce(() =>
		Promise.resolve({
			json: () => Promise.resolve({
				entries:
        [{
        		'id': 1,
        		'temp': 25.0,
        		'humidity': 45,
        		'dateTime': 'October 11, 2021 08:00:00',
        	},
        ],
			}),
		}));


	React.useState = jest.fn().mockReturnValue([{
		'id': 1,
		'temp': 25.0,
		'humidity': 45,
		'dateTime': 'October 11, 2021 08:00:00',
	}, {}]);
	const wrapper = await shallow(
		// <MemoryRouter initialEntries={[ '/random' ]}>
		<Routes/>
		// </MemoryRouter>
	);
    
	wrapper.update();
	// console.log(await wrapper.html());
	expect(wrapper.find(Dashboard)).toHaveLength(0);
	// expect(wrapper.find(Component404)).toHaveLength(1);
	expect(toJSON(wrapper)).toMatchSnapshot();
});