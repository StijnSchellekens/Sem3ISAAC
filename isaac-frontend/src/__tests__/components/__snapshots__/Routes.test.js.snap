// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`invalid paths should redirect to 404 1`] = `
<div>
  Loading...
</div>
`;
